internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +145,63 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
+	collectdLock.Lock()
+	defer collectdLock.Unlock()
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
+	if r.config.monitorConfig.MonitorConfigCore().IsCollectdBased() && !configuredCollectd {
+		r.setUpCollectdConfig(extensions)
+		if err := collectd.ConfigureMainCollectd(r.getCollectdConfig()); err != nil {
+			return nil, err
+		}
+		configuredCollectd = true
+	}
+
 	return monitor, nil
 }
+
+func (r *Receiver) setUpCollectdConfig(extensions map[configmodels.Extension]component.ServiceExtension) {
+	f := smartagentextension.NewFactory()
+	defaultCfg := f.CreateDefaultConfig().(*smartagentextension.Config)
+	r.config.collectdConfig = defaultCfg.CollectdConfig
+	// Do a lookup for any smartagent extensions to pick up common collectd options
+	// to be applied across instances of the receiver.
+	for c := range extensions {
+		if c.Type() != f.Type() {
+			continue
+		}
+
+		cfg, ok := c.(*smartagentextension.Config)
+		if !ok {
+			continue
+		}
+		r.config.collectdConfig = cfg.CollectdConfig
```

Comment:
I think this should operate at individual fields instead of writing over all the defaults

Reply:
For any field that does not have a user-specified value, the defaults would still correctly hold since the smartagent extension sets it, right?

Reply:
Sry, you're right.  I was conflating validation and custom unmarshalling.